src/Writer/TempFileGZIPWriter.php

```diff
@@ -50,7 +50,9 @@ public function finish(): void
         stream_copy_to_stream($this->tempFile, $file);
 
         fclose($file);
-        fclose($this->tempFile);
+
+        $t = $this->tempFile;
+        fclose($t);
```

Comment:
Why do we need to do it?

Reply:
it shut psalm up :-) 

ERROR: InvalidPropertyAssignmentValue - src/Writer/TempFileGZIPWriter.php:54:16 - $this->tempFile with declared type 'null|resource' cannot be assigned type 'closed-resource' (see https://psalm.dev/145)

Reply:
I guess that correct solution would be

```php
fclose($this->tempFile);
$this->tempFile = null;
```

Reply:
I'm not saying 'psalm' is necessarily correct. I'll change the attribute's phpdoc, I think that's better.


```   
   /**
     * Deflate buffered data
     */
    public function finish(): void
    {
        \assert($this->tempFile !== null);

        $file = fopen('compress.zlib://' . $this->filename, 'wb');
        rewind($this->tempFile);
        stream_copy_to_stream($this->tempFile, $file);

        fclose($file);

        fclose($this->tempFile); // Line 54
        $this->tempFile = null;
    }
}
```